server/src/main/java/io/crate/analyze/AlterTableAnalyzer.java

```diff
@@ -100,16 +105,18 @@ AnalyzedAlterTableRename analyze(AlterTableRename<Expression> node, SessionConte
         if (newIdentParts.size() > 1) {
             throw new IllegalArgumentException("Target table name must not include a schema");
         }
-
-        RelationName relationName;
+        
+        DocTableInfo tableInfo;
         if (node.blob()) {
-            relationName = RelationName.fromBlobTable(node.table());
+            var relationName = RelationName.fromBlobTable(node.table());
+            tableInfo = schemas.getTableInfo(relationName, Operation.ALTER_TABLE_RENAME);
         } else {
-            relationName = schemas.resolveRelation(node.table().getName(), sessionContext.searchPath());
+            var relationName = schemas.resolveRelation(node.table().getName(), sessionContext.searchPath());
+            tableInfo = new DocTableInfoBuilder(nodeCtx, relationName, clusterService.state()).build();
+            Operation.blockedRaiseException(tableInfo,Operation.ALTER_TABLE_RENAME);
```

Comment:
Why is it required to build a new tableInfo instance here instead of resolving it via schemas?

Reply:
To avoid to get a cached instance of `DocTableInfo`. The cached instances of`DocTableInfo` in `DocSchemaInfo` are not updated when a change on the custom publication metadata happens. `DocTableInfo` does not react to Metadata.custom but only to IndexMetadata. Would you prefer to adapt https://github.com/crate/crate/blob/master/server/src/main/java/io/crate/metadata/doc/DocSchemaInfo.java#L222 to also take the publication into account ?

Reply:
Yes I think we should adopt the invalidation.

Use-sites of `getTableInfo` shouldn't be aware of the caching or implementation details of the invalidation.

Reply:
I wonder if `SchemaInfo.invalidateTableCache` could help here